trieste/acquisition/function.py

```diff
@@ -130,6 +133,110 @@ def expected_improvement(model: ModelInterface, eta: tf.Tensor, at: QueryPoints)
+        :param model: The model over the specified ``dataset``.
+        :return: The MES function.
+        """
+        query_points = self._search_space.sample(self._grid_size)
+        query_points = tf.concat([dataset.query_points,query_points],0)
+        fmean, fvar = model.predict(query_points)
+        fsd = tf.math.sqrt(fvar)
+
+        def probf(x: tf.Tensor) -> tf.Tensor :  # Build empirical CDF
+            unit_normal = tfp.distributions.Normal(tf.cast(0, tf.float64), tf.cast(1, tf.float64))
+            log_cdf = unit_normal.log_cdf(- (x - fmean) / fsd)
+            return tf.exp(tf.reduce_sum(log_cdf, axis=0))
+
+        left = tf.reduce_min(fmean - 5 * fsd)
+        right = tf.reduce_max(fmean + 5 * fsd)
+
+        def binary_search(val: float) -> float:  # Fit Gumbel quantiles
+            return bisect(lambda x: probf(x) - val, left, right, maxiter=10000, xtol=0.00001)
+        
+        q1, med, q2 = map(binary_search, [0.25, 0.5, 0.75])
+
+        b = (q1 - q2) / (tf.math.log(tf.math.log(4. / 3.)) - tf.math.log(tf.math.log(4.)))
+        a = med + b * tf.math.log(tf.math.log(2.))
+
+        uniform_samples = tf.random.uniform([self._num_samples],dtype=tf.dtypes.float64)
+        gumbel_samples = -tf.math.log(-tf.math.log(uniform_samples)) * tf.cast(b, tf.float64) + tf.cast(a,tf.float64)
+
+        return lambda at: self._acquisition_function(model, gumbel_samples, at)
+
+    @staticmethod
+    def _acquisition_function(model: ModelInterface, samples: tf.Tensor, at: QueryPoints) -> tf.Tensor:
+        return max_value_entropy_search(model, samples, at)
+
+
+def max_value_entropy_search(model: ModelInterface, samples: tf.Tensor, at: QueryPoints) -> tf.Tensor:
+    r"""
+    Computes the information gain, i.e the change in entropy of p_min if we would evaluate x.
+
+    See the following for details:
+
+    ::
+
+        @article{wang2017max,
+          title={Max-value entropy search for efficient Bayesian optimization},
+          author={Wang, Zi and Jegelka, Stefanie},
+          journal={arXiv preprint arXiv:1703.01968},
+          year={2017}
+        }
+
+    :param model: The model of the objective function.
+    :param samples: Samples from p_min
+    :param at: The points for which to calculate the expected improvement.
+    :return: The expected improvement at ``at``.
+    """
+    fmean, fvar = model.predict(at)
+    fsd = tf.math.sqrt(fvar)
+    fsd = tf.clip_by_value(fsd, 1.0e-8, tf.float64.max) # clip below to improve numerical stability
+    
+    normal = tfp.distributions.Normal(tf.cast(0,tf.float64), tf.cast(1,tf.float64))
+    gamma = (samples - fmean) / fsd
+
+    minus_cdf = 1 - normal.cdf(gamma)
+    minus_cdf = tf.clip_by_value(minus_cdf, 1.0e-8, 1) # clip below to improve numerical stability
+    f_acqu_x = -gamma * normal.prob(gamma) / (2 * minus_cdf) - tf.math.log(minus_cdf)
+    f_acqu_x = tf.math.reduce_mean(f_acqu_x, axis=1)
+        
+    return tf.reshape(f_acqu_x,[-1, 1])
```

Comment:
```suggestion
    return tf.math.reduce_mean(f_acqu_x, axis=1, keepdims=True)
```
to allow for batch dimensions?

Reply:
Good spot!

Reply:
I'm keen to do a batch version of this based on some of my PhD work that I am typing up. I'll do this as another PR later. I think it will require a bit more work to support batch optimisation.